syntax = "proto3";

package potechi;

service Potechi {
  rpc Vote (stream VoteRequest) returns (stream VoteReply) {}
  rpc Counting (CountingRequest) returns (CountingReply) {}
}

message VoteRequest {
  repeated int32 flavor_numbers = 1;
  string voter_name = 2;
}

message VoteReply {
  map<string, int32> result = 1;
}

message CountingRequest {
  string voter_name = 1;
}

message CountingReply {
  double contribution_rate = 1;
  string message = 2;
}